modules/templates/handler.rb

```diff
@@ -7,23 +7,24 @@ module Proxy::Templates
   class Handler < ::Proxy::HttpRequest::ForemanRequest
     include Proxy::Log
 
-    def get_template(kind, token, static = false)
-      opts = { :token => token,
-               :url   => Proxy::Templates::Plugin.settings.template_url
-      }
-      opts[:static] = static if static
+    def get_template(kind, params, headers)
+      opts = params.merge(:url => Proxy::Templates::Plugin.settings.template_url)
       request = request_factory.create_get("/unattended/#{kind}", opts)
+      headers.each do |k, v|
+        request.add_field(k, v)
```

Comment:
Probably worth modifying `ForemanRequestFactory` to accept headers as a parameter to `create_` methods (https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb#L16).